File: app/src/main/java/velkus/moneybagsbudgeting/storage/DatabaseFactory.java
package velkus.moneybagsbudgeting.storage;

import android.arch.persistence.room.Room;
import android.content.Context;

public class DatabaseFactory {
    private static AppDatabase instance = null;

    public static AppDatabase getInstance(Context context) {
        if (instance != null) {
            return instance;
        }

        synchronized (AppDatabase.class) {
            if (instance != null) {
                return instance;
            } else {
                instance = Room.databaseBuilder(context, AppDatabase.class, "moneybags.db")
                        .allowMainThreadQueries()
                        .build();
                return instance;
            }
        }
    }
}

File: app/src/main/java/velkus/moneybagsbudgeting/storage/DatabaseFactory.kt
package velkus.moneybagsbudgeting.storage

import android.arch.persistence.room.Room
import android.content.Context

object DatabaseFactory {
    private var instance: AppDatabase? = null

    fun getInstance(context: Context): AppDatabase {
        if (instance != null) {
            return instance!!
        }

        synchronized(AppDatabase::class.java) {
            return if (instance != null) {
                instance!!
            } else {
                instance = Room.databaseBuilder(context, AppDatabase::class.java, "moneybags.db")
                        .addMigrations(Migrations.MIGRATION_1_2)
                        .allowMainThreadQueries()
                        .build()
                instance!!
            }
        }
    }
}